session/middleware/start_session_test.go

```diff
@@ -0,0 +1,368 @@
+package middleware
+
+import (
+	"context"
+	nethttp "net/http"
+	"net/http/httptest"
+	"net/url"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/goravel/framework/contracts/filesystem"
+	contractshttp "github.com/goravel/framework/contracts/http"
+	contractsession "github.com/goravel/framework/contracts/session"
+	"github.com/goravel/framework/contracts/validation"
+	configmocks "github.com/goravel/framework/mocks/config"
+	"github.com/goravel/framework/session"
+	"github.com/goravel/framework/support/file"
+)
+
+func testHttpSessionMiddleware(next nethttp.Handler, mockConfig *configmocks.Config) nethttp.Handler {
+	return nethttp.HandlerFunc(func(w nethttp.ResponseWriter, r *nethttp.Request) {
+		session.ConfigFacade = mockConfig
+		session.Facade = session.NewManager(mockConfig)
+		StartSession()(NewTestContext(r.Context(), next, w, r))
+	})
+}
+
+func mockConfigFacade(mockConfig *configmocks.Config) {
+	mockConfig.On("GetString", "session.driver").Return("file").Once()
+	mockConfig.On("GetInt", "session.lifetime").Return(60).Times(3)
+	mockConfig.On("GetString", "session.files").Return("sessions").Once()
+	mockConfig.On("GetString", "session.cookie").Return("goravel_session").Once()
+	mockConfig.On("GetString", "session.path").Return("/").Once()
+	mockConfig.On("GetString", "session.domain").Return("").Once()
+	mockConfig.On("GetBool", "session.secure").Return(false).Once()
+	mockConfig.On("GetBool", "session.http_only").Return(true).Once()
+	mockConfig.On("GetString", "session.same_site").Return("").Once()
+	mockConfig.On("Get", "session.lottery").Return([]int{100, 100}).Once()
+}
+
+func TestStartSession(t *testing.T) {
+	mockConfig := &configmocks.Config{}
+	mockConfigFacade(mockConfig)
+
+	server := httptest.NewServer(testHttpSessionMiddleware(nethttp.HandlerFunc(func(w nethttp.ResponseWriter, r *nethttp.Request) {}), mockConfig))
```

Comment:
We can add some data to the session in the handler, and then test if the data can be fetched successfully in another request.